File: src/Core/Entities/OrderAggregate/Order.cs
namespace Core.Entities.OrderAggregate;

public class Order : BaseEntity
{
   public DateTime OrderDate { get; set; } = DateTime.UtcNow;
   public string BuyerEmail { get; set; } = default!;
   public DeliveryMethod DeliveryMethod { get; set; } = null!;
   public ShippingAddress ShippingAddress { get; set; } = null!;
   public PaymentSummary PaymentSummary { get; set; } = null!;
   public List<OrderItem> OrderItems { get; set; } = [];
   public decimal Subtotal { get; set; }
   public OrderStatus Status { get; set; } = OrderStatus.Pending;
   public string PaymentIntentId { get; set; } = default!;

   public decimal GetTotal()
   {
      return Subtotal + DeliveryMethod.Price;
   }
}

File: src/Client/src/app/shared/models/order.ts

export interface Order {
  id: number;
  orderDate: string;
  buyerEmail: string;
  deliveryMethod: string;
  shippingAddress: ShippingAddress;
  paymentSummary: PaymentSummary;
  orderItems: OrderItem[];
  shippingPrice: number;
  subtotal: number;
  total: number;
  status: string;
  paymentIntentId: string;

}

export interface ShippingAddress {
  name: string
  line1: string
  line2?: string
  city: string
  state: string
  postalCode: string
  country: string
}

export interface PaymentSummary {
  last4: number
  brand: string
  expMonth: number
  expYear: number
}

export interface OrderItem {
  productId: number
  productName: string
  pictureUrl: string
  price: number
  quantity: number
}


export interface OrderToCreate {
 cartId: string;
 deliveryMethodId: number;
 shippingAddress: ShippingAddress;
 paymentSummary: PaymentSummary;
}
